internal/handlers/tigris/msg_findandmodify.go

```diff
@@ -130,30 +132,28 @@ func (h *Handler) MsgFindAndModify(ctx context.Context, msg *wire.OpMsg) (*wire.
 				if _, err = updateDocument(ctx, dbPool, &qp, upsert); err != nil {
 					return nil, err
 				}
-			}
-		}
 
-		var resultDoc *types.Document
-		if params.ReturnNewDocument || len(resDocs) == 0 {
```

Comment:
`len(resDocs) == 0` case returns early, see L103. So this is unreachable hence removing